### ISSUE TITLE
feat: Add InstructPipeline::generate() for autoregressive text generation (SSC CLI-002)

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#246]
## Summary

Add a first-class `generate()` method to `InstructPipeline` for autoregressive text generation with LoRA adapters. This enables bashrs CLI-002 (wire `bashrs explain/fix → chat model`).

## Background

entrenar already has all building blocks:
- `Transformer::forward()` for logits
- `forward_hidden()` helper
- LoRA layers
- `HfTokenizer` with encode/decode
- `format_chat_prompt` for ChatML
- Working generation in `examples/finetune_real.rs` (greedy decoding)

Missing: proper API method with temperature, top-k, and max_new_tokens support.

### 1. `InstructPipeline::generate(prompt, config) -> String`

```rust
pub struct GenerateConfig {
    pub max_new_tokens: usize,  // default: 256
    pub temperature: f32,       // default: 0.7
    pub top_k: usize,           // default: 50
    pub stop_tokens: Vec<u32>,  // default: [eos_id]
}

impl InstructPipeline {
    pub fn generate(&self, prompt: &str, config: &GenerateConfig) -> String;
    pub fn generate_chat(&self, messages: &[(String, String)], config: &GenerateConfig) -> String;
}
```

### 2. CPU-only forward path

The generation loop (from finetune_real.rs pattern):
1. Tokenize prompt
2. `forward()` → logits for last position
3. Sample (temperature + top-k)
4. Append token, repeat until EOS/max
5. Decode

### 3. Load from checkpoint

`InstructPipeline::from_checkpoint(model_dir: &Path)` — load base model + LoRA adapter from saved checkpoint.

## Acceptance Criteria

- [ ] `InstructPipeline::generate()` works on CPU
- [ ] `InstructPipeline::generate_chat()` formats ChatML and generates
- [ ] `InstructPipeline::from_checkpoint()` loads saved model
- [ ] Temperature and top-k sampling
- [ ] Unit tests for generation loop

## Context

Needed for: bashrs shell-safety-inference.md Phase 4 CLI-002
Spec: `paiml/bashrs` docs/specifications/shell-safety-inference.md Section 8.1
